### PR TITLE
Add CodeCov and created a GitHub Action For code Coverage

### PR DESCRIPTION
**Pull request by wvanderp**
This PR will add back code coverage and uploading to CodeCov.

I created a Github action that creates code coverage and uploads it to CodeCov. It runs every pull request to master and every push to master.

I've tested it with this PR: https://github.com/wvanderp/wtf_wikipedia/pull/2 . It might be a bit of a bad example because I commented out a test that keeps failing in master during the last run then I changed master but did not run coverage and thus the change are picked up here. but this should not be a problem because you should not commit to master.

you also see a big comment from CodeCov. This is added by the codecov github app @ https://github.com/apps/codecov .
It also adds checks that alert users when code in therePR is not covered by tests (src: https://docs.codecov.io/docs/github-checks-beta)

You can also configure what codecov does with a yaml file. see https://docs.codecov.io/docs/codecov-yaml .

So to summarize this PR will add  back coverage. But the integration with github needs some work from your side.

-wvanderp

### CODE REVIEW

**Comment by spencermountain** [comment]
hey this is really helpful! 

Thank you wouter.

**Comment by spencermountain** [comment]
worked!
image
👏
